### ISSUE TITLE
Crash in getMeaningFromRightHandSideOfImportEquals during getReferencesAtPosition

### ISSUE THREAD

**Issue by RyanCavanaugh** [Microsoft/TypeScript#20546]
Stack
```
Error: Debug Failure. False expression.
    at getMeaningFromRightHandSideOfImportEquals (C:\github\TypeScript\lib\typescript.js:77553:18)
    at Object.getMeaningFromLocation (C:\github\TypeScript\lib\typescript.js:77532:20)
    at getReferencedSymbolsForSymbol (C:\github\TypeScript\lib\typescript.js:83388:79)
    at Object.getReferencedSymbolsForNode (C:\github\TypeScript\lib\typescript.js:83307:24)
    at findAllReferencedSymbols (C:\github\TypeScript\lib\typescript.js:83143:43)
    at Object.findReferencedEntries (C:\github\TypeScript\lib\typescript.js:83132:36)
    at getReferences (C:\github\TypeScript\lib\typescript.js:97907:41)
    at Object.getReferencesAtPosition (C:\github\TypeScript\lib\typescript.js:97890:20)
    at Object.<anonymous> (C:\github\ts-codefix-fuzz\bin\run.js:126:21)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Function.Module.runMain (module.js:605:10)
    at startup (bootstrap_node.js:158:16)
    at bootstrap_node.js:575:3
```
Unreduced repro:
```ts
function  (  function  :  string ,  y ,  z  =  3 ,  q ? ,  z ) { }; import  j  =  NM . q ; class  C  {  public   m ;  private readonly  function  j  =  null ; constructor( y ,  m ,  ... q ) {}} namespace  NM  {  f ( 10 ); new  C ( 100 ,  3 ,  undefined );  export  const  q  =  2 ;  declare  function  g ( p ):  any  = {  x  = {  m ( n ) { },  p :  y     t ;  abstract   q ;  static   =>  [] };
```

**Comment by RyanCavanaugh** [comment]
Minimal repro testcase
```ts
/// <reference path="fourslash.ts" />

//// import j = NM./**/ q;
//// namespace NM { export const q;

goTo.marker();
verify.noReferences();
```
Observed constraints which I would not believe had I not seen them myself:
 * Space between the `.` and `q` at the marker position
 * *and* the `namespace` block has to be unclosed
 * *and* the `const` has to be uninitialized

**Comment by andy-ms** [comment]
Actually, the only thing needed to reproduce the error is the space in `N. q`. (It's also necessary for the symbol `N.q` to exist, else we exit early with no error.)
```ts
/// <reference path="fourslash.ts" />

////import j = N./**/ q;
////namespace N { export const q = 0; }
goTo.marker("");
verify.noReferences();
```
The problem is that since there is a space, `getTouchingPropertyName` now returns the full `QualifiedName` `N.q` instead of `q` (which we're not *touching*). Fixed in #21779.
